### PR TITLE
Seed v2.0.0 into `next`

### PR DESCRIPTION
**Pull request by bryanculver**
Gotta start somewhere.

- Seeds release notes
- Starts an upgrade guide
- Bumps `pyproject.toml`

### CODE REVIEW

**Review by gsnider2195 — APPROVED** [review]

**Review by glennmatthews — APPROVED** [review]

**Comment by bryanculver** [comment]
@glennmatthews Thoughts on this to keep CI happy for now:

https://github.com/nautobot/nautobot/pull/2788/files#diff-6c6cc147c8188d94be8fd3b11345aa4bfef52a760e0961643b6b84d0c26da43aL175-R174

File: nautobot/core/settings.py [hunk 1]
```diff
@@ -170,10 +170,8 @@
 REST_FRAMEWORK_VERSION = VERSION.rsplit(".", 1)[0]  # Use major.minor as API version
 current_major, current_minor = REST_FRAMEWORK_VERSION.split(".")
 # We support all major.minor API versions from 1.2 to the present latest version.
-# This will need to be elaborated upon when we move to version 2.0
 # Similar logic exists in tasks.py, please keep them in sync!
-assert current_major == "1", f"REST_FRAMEWORK_ALLOWED_VERSIONS needs to be updated to handle version {current_major}"
-REST_FRAMEWORK_ALLOWED_VERSIONS = [f"{current_major}.{minor}" for minor in range(2, int(current_minor) + 1)]
+REST_FRAMEWORK_ALLOWED_VERSIONS = ["1.2", "1.3", "1.4", "1.5"] + [f"{current_major}.{minor}" for minor in range(0, int(current_minor) + 1)]
```
Comment (bryanculver) [inline nautobot/core/settings.py:174]: @glennmatthews Thoughts on this to keep CI happy for now?

File: tasks.py [hunk 2]
```diff
@@ -627,8 +627,9 @@ def check_schema(context, api_version=None):
         nautobot_version = get_nautobot_version()
         # logic equivalent to nautobot.core.settings REST_FRAMEWORK_ALLOWED_VERSIONS - keep them in sync!
         current_major, current_minor = nautobot_version.split(".")[:2]
-        assert current_major == "1", f"check_schemas version calc must be updated to handle version {current_major}"
```
Comment (glennmatthews) [inline tasks.py:630]: Same here, let's make this check for "2" rather than removing it.

**Review by glennmatthews — APPROVED** [review]

File: poetry.lock [hunk 3]
```diff
@@ -210,7 +210,7 @@ optional = false
 python-versions = ">=3.5.0"
 
 [package.extras]
-unicode-backport = ["unicodedata2"]
+unicode_backport = ["unicodedata2"]
```
Comment (HanlinMiao) [inline poetry.lock:213]: Whats the reason for the change from "-" to "_"?

**Review by HanlinMiao — APPROVED** [review]

File: poetry.lock (hunk 3 above)
Comment (glennmatthews) [reply poetry.lock:213]: Hmm, check the `poetry` version. I think this may be a change in 1.2.2?

Comment (HanlinMiao) [reply poetry.lock:213]: Yeah I see this is where they canonicalized extra names. https://github.com/python-poetry/poetry/pull/6541/

Comment (bryanculver) [reply poetry.lock]: 🤦

**Pull request merged**